test/python/test_qi.py

```diff
@@ -134,6 +146,117 @@ def test_concurrence(self):
                      concurrence(rho4) == 0.5)
         self.assertTrue(test_pass)
 
+    def test_concurrence_not_two_qubits(self):
```

Comment:
This actually reveals that maybe the `concurrence()` function should be updated. 

First, it should only be defined for 2 qubits. That is, the Exception message should be updated to reflect this: ("Concurence is only defined for more than two qubits").

Second, we should update the docstring to say the argument can both be a pure state and a density matrix.
```
    Args:
        state (np.array): a quantum state (1x4 array) or a density matrix (4x4) array
```

Reply:
Done